components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -1181,6 +1277,37 @@ void triggerPostRevokeListeners(OAuthRevocationRequestDTO revokeRequestDTO,
         }
     }
 
+    private void triggerPreApplicationTokenRevokeListeners(OAuthAppRevocationRequestDTO revokeRequestDTO) {
+
+        OAuthEventInterceptor oAuthEventInterceptorProxy = OAuthComponentServiceHolder.getInstance()
+                .getOAuthEventInterceptorProxy();
+        if (oAuthEventInterceptorProxy != null && oAuthEventInterceptorProxy.isEnabled()) {
+            Map<String, Object> paramMap = new HashMap<>();
+            try {
+                oAuthEventInterceptorProxy.onPreTokenRevocationByApplication(revokeRequestDTO, paramMap);
+            } catch (IdentityOAuth2Exception e) {
+                LOG.error("Error occurred when triggering pre revocation listener.", e);
+            }
+        }
+    }
+
+    private void triggerPostApplicationTokenRevokeListeners(OAuthAppRevocationRequestDTO revokeRequestDTO,
+                                                            OAuthRevocationResponseDTO revokeRespDTO,
+                                                            List<AccessTokenDO> accessTokenDOs) {
+
+        OAuthEventInterceptor oAuthEventInterceptorProxy = OAuthComponentServiceHolder.getInstance()
+                .getOAuthEventInterceptorProxy();
+        if (oAuthEventInterceptorProxy != null && oAuthEventInterceptorProxy.isEnabled()) {
+            Map<String, Object> paramMap = new HashMap<>();
+            try {
+                oAuthEventInterceptorProxy.onPostTokenRevocationByApplication(revokeRequestDTO, revokeRespDTO,
+                        accessTokenDOs, paramMap);
+            } catch (IdentityOAuth2Exception e) {
+                LOG.error("Error occurred when triggering post revocation listener.", e);
```

Comment:
Any reason for not rethrowing?

Reply:
Addressed in [9584b3f](https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1627/commits/9584b3fe41b6715d60d1759f2a6a8b453fdfc65a)